py-polars/tests/unit/operations/rolling/test_rolling.py

```diff
@@ -589,6 +589,36 @@ def test_rolling_cov_corr() -> None:
     assert res["corr"][:2] == [None] * 2
 
 
+def test_rolling_cov_corr_nulls() -> None:
+    df1 = pl.DataFrame(
+        {"a": [1.06, 1.07, 0.93, 0.78, 0.85], "lag_a": [1.0, 1.06, 1.07, 0.93, 0.78]}
+    )
+    df2 = pl.DataFrame(
+        {
+            "a": [1.0, 1.06, 1.07, 0.93, 0.78, 0.85],
+            "lag_a": [None, 1.0, 1.06, 1.07, 0.93, 0.78],
+        }
+    )
+
+    val_1 = df1.select(
+        pl.rolling_corr("a", "lag_a", window_size=10, min_periods=5, ddof=1).tail(1)
+    ).item()
+    val_2 = df2.select(
+        pl.rolling_corr("a", "lag_a", window_size=10, min_periods=5, ddof=1).tail(1)
+    ).item()
+
+    assert val_1 == val_2
+
+    val_1 = df1.select(
+        pl.rolling_cov("a", "lag_a", window_size=10, min_periods=5, ddof=1).tail(1)
+    ).item()
+    val_2 = df2.select(
+        pl.rolling_cov("a", "lag_a", window_size=10, min_periods=5, ddof=1).tail(1)
+    ).item()
+
+    assert val_1 == val_2
```

Comment:
Can you also test the actual value here?

Reply:
I changed the test as suggested and pushed. However, I also spent some time trying to put together a hypothesis test that would cross check these corr and cov functions against numpy. I could not get it to pass, and have an example frame which yields correlation > 1.0 :-/

df = pl.DataFrame(
    {
        "a": [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0],
        "b": [101.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.000061, 0.0],
    }
)

df_corr = df.select(
    pl.rolling_corr("a", "b", window_size=7, min_periods=5, ddof=1)
)

I don't have time to push more on this right now (or even this week maybe). But I will log a separate issue.